src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicSchemaToViewTransformer.java

```diff
@@ -0,0 +1,202 @@
+            createTypeTable(productConfig.getPackageName()),
+            namer,
+            JavaFeatureConfig.newBuilder().enableStringFormatFunctions(false).build());
+
+    StaticLangApiMessageFileView apiFile;
+    for (Map.Entry<String, Schema> entry : context.getDocument().schemas().entrySet()) {
+      apiFile = generateSchemaFile(context, entry.getKey(), entry.getValue(), docSymbolTable);
+      surfaceSchemas.add(apiFile);
+    }
+    return surfaceSchemas;
+  }
+
+  private ModelTypeTable createTypeTable(String implicitPackageName) {
+    return new ModelTypeTable(
+        new JavaTypeTable(implicitPackageName),
+        new JavaModelTypeNameConverter(implicitPackageName));
+  }
+
+  private StaticLangApiMessageFileView generateSchemaFile(
+      DiscoGapicInterfaceContext context,
+      String schemaName,
+      Schema schema,
+      SymbolTable docSymbolTable) {
+    StaticLangApiMessageFileView.Builder apiFile = StaticLangApiMessageFileView.newBuilder();
+    // Escape any schema's field names that are Java keywords.
+    SymbolTable schemaSymbolTable = SymbolTable.fromSeed(JavaNameFormatter.RESERVED_IDENTIFIER_SET);
+
+    apiFile.templateFileName(SCHEMA_TEMPLATE_FILENAME);
+
+    apiFile.schema(generateSchemaClass(context, schemaName, schema, schemaSymbolTable));
+
+    String outputPath = pathMapper.getOutputPath(null, context.getProductConfig());
+    apiFile.outputPath(outputPath + File.separator + schemaName + ".java");
+
+    // must be done as the last step to catch all imports
+    apiFile.fileHeader(fileHeaderTransformer.generateFileHeader(context));
+
+    return apiFile.build();
+  }
+
+  private StaticLangApiMessageView generateSchemaClass(
+      DiscoGapicInterfaceContext context,
+      String schemaName,
+      Schema schema,
+      SymbolTable schemaSymbolTable) {
+    addApiImports(context);
+
+    StaticLangApiMessageView.Builder schemaView = StaticLangApiMessageView.newBuilder();
+
+    schemaView.typeName(schemaName);
+    schemaView.type(schema.type());
+    // TODO(andrealin): apply Java naming format.
+    // TODO(andrealin): use symbol table to make sure Schema names aren't Java keywords.
+    schemaView.className(schemaName);
+    schemaView.defaultValue(schema.defaultValue());
+
+    // Map each property name to the Java typeName of the property.
+    List<SimplePropertyView> properties = new LinkedList<>();
+    for (Map.Entry<String, Schema> propertyEntry : schema.properties().entrySet()) {
+      String propertyName = schemaSymbolTable.getNewSymbol(propertyEntry.getKey());
+      Schema property = propertyEntry.getValue();
+      SimplePropertyView.Builder simpleProperty =
+          SimplePropertyView.newBuilder().name(propertyName).repeated(property.repeated());
+      simpleProperty.typeName(typeToJavaType(property));
+
+      // Property class name is Capitalized
+      simpleProperty.capitalizedName(
+          propertyName.substring(0, 1).toUpperCase() + propertyName.substring(1));
```

Comment:
Don't do string casing changes in transformers - that's the job of the surface namer. Since you don't have a namer yet, at a minimum, use code like this to compose new identifiers from existing ones:

https://github.com/googleapis/toolkit/blob/2853924324aeef3cabfe9c9c50a66009ead1f0b3/src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java#L282